text/0000-lifecycle-all.md

```diff
@@ -0,0 +1,109 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle All Binary
+- Start Date: 2020-01-21
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+### `/cnb/lifecycle/all`
+A new lifecycle binary with the name `lifecycle/all` will be included in each released lifecycle archive. Within a builder image, it will be found at the path `/cnb/lifecycle/all`. When invoked, it will run the following phases sequentially:
+
+* detector
+* analyzer
+* restorer
+* builder
+* exporter
+
+Each of these phases will continue to be available individually as provided by the current lifecycle.
+
+### Platform API 0.3
+The existence of `all` will constitute a non-breaking but substantive change to the Platform API, bringing us to Platform API 0.3 . Although platform will interpret this as a breaking change, we do not have a mechanism for indicating non-breaking platform API change pre 1.0. Bumping the platform API number is necessary so that platforms like `pack` know whether or no they can use this feature.
+
+# Motivation
+[motivation]: #motivation
+
+This new binary supports two main goals.
+
+### Goal 1: faster `pack` builds
+By creating a single container and invoking `/cnb/lifecycle/all` instead of creating a container per lifecycle phase, pack can shave approximate `6s` from the execution time of `pack build`.
+
+`pack` may not always want to do this (see https://github.com/buildpacks/rfcs/pull/43 for an explanation of why this is not always ideal). However, it could be very beneficial to users that are wiling to trust their builder images.
+
+### Goal 2: Easier integration of the lifecycle into other platforms and CI tools
```

Comment:
This is a more compelling argument, when compared to pure performance which I think can be enhanced in other ways.

Reply:
Even if we want this feature mainly to accomplish `Goal 2` I think `pack` should use it. The ux and the reliability of `/cnb/lifecycle/all` will be improved if `pack` uses it in anger.

I can reverse the order of the goals if we think this is the more compelling one.

Reply:
Not necessary.